### DISCUSSION TITLE
Modifying and Using Redash commercially

### DISCUSSION THREAD

**Discussion by internetchris** [getredash/redash#6105] [Q&A] [answered]
Hey group,

My team has been tasked with building a product that would be very similiar to Redash. I would prefer to not re-invent the wheel and I like the idea of supporting a community project like this, while saving time and headache. Having said that, I've read the terms of use on the redash.io site and also the markdown file here: https://github.com/getredash/redash/blob/master/LICENSE

The redash.io site has as strict service policy restricting the resale of the service, I think the key term being "service".  Where'as if we host our own servers and fork the code/ modify it for our specific business needs it seems permissible under the github LICENSE markdown. I'm noting the "green checkbox" next to the commercial usage indicator in that markdown file. 

I'm not looking to interpret the policy's to my benefit. I'd like to know in no uncertain terms if using redash source code for commercial benefit is permissible.  I'd also like to applaud this community for what looks to be a very awesome tool!

**Comment by internetchris** [comment]
@arikfr - I'd love your input here if you have a minute. Thanks for all you do.

**Comment by justinclift** [comment]
Yep, go for it.  You're completely fine to use it for commercial purposes.

If it goes well, hopefully your team will be involved in contributing future development to the main Redash project as well. :smile:

**Comment by arikfr** [comment, marked as answer]
Redash is an open source project and what you're planning to do sounds like within what the license allows and encourages. 

The terms on the website were for the hosted service which is no longer available and irrelevant to the OSS project. We should remove them.

**Comment by internetchris** [comment]
@arikfr , @justinclift  - thanks for the response. Sorry for the delay. I also had to jump through our own company legal hoops to get approval as well. Having said that, consider us contributors! I'm queueing up tickets to have my team start assessing what we need and how we can contribute. Theme'ing is at the top of the list.

**Reply by justinclift** [reply]
Heh Heh Heh.  Right at this moment, your front end developers (javascript, typescript) would be welcome to help us update several of the javascript related dependencies.

For good measure, are they familiar with [Cypress](https://www.cypress.io)?

That's the End-to-End (E2E) testing framework we use.  It's not hard to get the hang of though, so either way they'll be fine.

Documentation wise, you're probably best to point them to the wiki:

https://github.com/getredash/redash/wiki

The "Local development setup" page on the wiki is known to work fine, whereas the (very outdated) development documentation on the official knowledge base is very broken.  Yeah, we need to update the official docs.  It'll happen eventually. :smile:
